File: Python/e034.py
"""
Problem 34: Digit factorials
145 is a curious number, as 1! + 4! + 5! = 1 + 24 + 120 = 145.
Find the sum of all numbers which are equal to the sum of the factorial
of their digits.
Note: as 1! = 1 and 2! = 2 are not sums they are not included.
"""
import math


def sumfac(num):
    digits = [int(i) for i in str(num)]
    return sum([math.factorial(x) for x in digits])


s = 0
for n in range(3, 10000000):
    if n == sumfac(n):
        print(n)
        s += n
print(s)

File: e034.js
function factorial(n){
    let p = 1;
    for (let i = n; i >= 1; i--){
        p *= i;
    }
    return p;
}

function sumfac(num){
    const digits = Array.from(num.toString());
    let sum = 0;
    digits.forEach(x => sum += factorial(x));
    // console.log(digits);
    return sum;
}

let s = 0;
for (let n = 3; n < 10000000; n++){
    if (n === sumfac(n)){
        console.log(n);
        s += n;
    }
}
console.log(s);